pkg/metrics/metrics.go

```diff
@@ -582,12 +584,31 @@ func prometheusToken(c client.Client) (*string, error) {
 	}
 
 	if len(token) == 0 {
-		return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA")
+		// If serviceaccount token doesn't exist, try to use token request to get a token
+		token, err = RequestPrometheusServiceAccountAPIToken(c)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA due to %v", err)
+		}
 	}
 
 	return &token, nil
 }
 
+// RequestPrometheusServiceAccountAPIToken returns a time-bound (24hr) API token for the prometheus service account.
+func RequestPrometheusServiceAccountAPIToken(c client.Client) (string, error) {
+	expirationSeconds := int64(10 * time.Minute / time.Second)
```

Comment:
Can we change this to 24 hours again or do you intend it to be 10 minutes ahead? If 10 mins then will need to update the function comment as well.

Reply:
Yes, the time will be changed to 24 hours. 10 mins here just for testing.

Reply:
Change back to 24 hours